packages/modeling/src/maths/mat4/rotate.js

```diff
@@ -14,14 +15,14 @@ const copy = require('./copy')
  */
 const rotate = (out, matrix, radians, axis) => {
   let [x, y, z] = axis
-  let len = Math.hypot(x, y, z)
+  const lengthSquared = x * x + y * y + z * z
 
-  if (Math.abs(len) < 0.000001) {
+  if (Math.abs(lengthSquared) < EPSILON) {
```

Comment:
I suggest that we use the EPS for normals, then we know that this function works for calculations with normals.